pkg/actuators/machine/instances.go

```diff
@@ -134,3 +134,33 @@ func getNetworkID(network powervsproviderv1.PowerVSResourceReference, client pow
 
 	return nil, fmt.Errorf("failed to find a network ID")
 }
+
+func getServiceInstanceID(serviceInstance powervsproviderv1.PowerVSResourceReference, client powervsclient.Client) (*string, error) {
+	if serviceInstance.ID != nil {
+		klog.Infof("Found the service %v", serviceInstance)
+		return serviceInstance.ID, nil
+	} else if serviceInstance.Name != nil {
+		serviceInstances, err := client.GetCloudServiceInstances()
+		if err != nil {
+			klog.Errorf("failed to get serviceInstances, err: %v", err)
+			return nil, err
+		}
+		// Create a instanceMap check whether there exist two service instances with same name
+		instanceMap := make(map[string]string)
+		for _, si := range serviceInstances {
+			if val, ok := instanceMap[si.Name]; ok {
+				errStr := fmt.Errorf("there exist two service instance ID with guid %s, %s with same name %s", val, si.Guid, si.Name)
+				klog.Errorf(errStr.Error())
+				return nil, errStr
+			}
+			instanceMap[si.Name] = si.Guid
+		}
```

Comment:
Lets not really worry about the service instances which aren't really matching the name passed by the user, lets keep it simple: if multiple service instances exists with the name mentioned in the spec then fail or else proceed.